angularjs-portal-frame/src/main/webapp/js/main/main-directives.js

```diff
@@ -18,12 +18,35 @@
     }
   });
 
-  app.directive('search', function() {
+  app.directive('search', ['miscService', 'marketplaceService', '$location', function(miscService, marketplaceService, $location) {
     return {
       restrict : 'E',
-      templateUrl : 'partials/search.html'
+      templateUrl : 'partials/search.html',
+      controller: function($scope) {
+          $scope.initialFilter = '';
+          $scope.filterMatches = [];
+          $scope.portletListLoading = true;
+          marketplaceService.getPortlets().then(function(data){
```

Comment:
There is opportunity to utilize the ng-storage of the marketplace here. We should first check the cache before pulling from server side. Also remember to check the session key matches. See marketplace controller for details. Might also just move that check to the service.